src/trajectory.jl

```diff
@@ -146,25 +149,28 @@ transition(
 ### Static trajecotry with fixed leapfrog step numbers.
 ###
 
-struct StaticTrajectory{S<:AbstractTrajectorySampler,I<:AbstractIntegrator} <: AbstractTrajectory{I}
+struct StaticTrajectory{S<:AbstractTrajectorySampler, I<:AbstractIntegrator} <: AbstractTrajectory{I}
     integrator  ::  I
     n_steps     ::  Int
 end
 
-Base.show(io::IO, τ::StaticTrajectory{S,I}) where {I,S<:EndPointTS} =
+function Base.show(io::IO, τ::StaticTrajectory)
```

Comment:
We do need the type info for correct dispatch here.

Reply:
Maybe you are intended to do what you did for `HMCDA`?

Reply:
```suggestion
function Base.show(io::IO, τ::StaticTrajectory{<:EndPointTS})
```